### DISCUSSION TITLE
ArgoCD EKS with AWS Cognito login configuration

### DISCUSSION THREAD

**Discussion by leandroschuster** [argoproj/argo-cd#7955] [Q&A] [answered]
Hello,

I am trying to integrate our ArgoCD that is deployed in Amazon EKS with a Cognito User Pool but I keep getting the error:

Failed to query provider "https://$COGNITO_POOL_URL/$Cognito_USER_POOL_ID": 400 Bad Request:"

My configuration in argocd-cm ConfigMap is as follows:

oidc.config: |
    name: Cognito
    issuer: https://$COGNITO_POOL_URL/$Cognito_USER_POOL_ID
    clientID: $cognito_client_id
    clientSecret: $cognito_client_secret
    requestedScopes:
    - openid
    - profile
    - phone
    - email

This user pool is already working for Jenkins. The code above is redacted for security (No variables in the actual code)

Has anyone configured ArgoCD to use AWS Cognito already? What am I missing?

Thank you.

**Comment by leandroschuster** [comment, marked as answer]
I managed to get it to work. Basically I had to use the Cognito discover URL (https://cognito-idp.$REGION.amazonaws.com/$USER_POOL_ID)

So my config is looking like this if anyone bumps into the same situation:

argocd-cm:

```
oidc.config: |
    name: Cognito
    issuer: https://cognito-idp.$REGION.amazonaws.com/$USER_POOL_ID
    clientID: $CLIENT_ID
    clientSecret: $CLIENT_SECRET
    requestedScopes: ["openid", "profile", "email"]
    requestedIDTokenClaims: {"groups": {"essential": true}}
```

argocd-rbac-cm
```
policy.csv: |
    g, admin, role:org-admin
  scopes: '[cognito:groups]'
```

**Comment by vavdoshka** [comment]
was really great starting point for me, thanks.
I've compiled a detailed guid on this topic: https://gist.github.com/vavdoshka/f180bc49886728d9f051d9c4db045f9e
